### ISSUE TITLE
Wordpress Bitnami - Theme Licencing Issue - Cariera

### ISSUE THREAD

**Issue by doduffy88** [bitnami/vms#1463]
### Platform

Google Cloud Platform

### bndiagnostic ID [know more about bndiagnostic ID](https://docs.bitnami.com/general/how-to/understand-bndiagnostic/)

81341a4a-a01c-54aa-d6e9-7e6350fbdc3d

### bndiagnostic output

===== Begin of bndiagnostic tool output =====

 
    ? Wordpress: Found possible issues
    ✓ Processes: No issues found
    ✓ Apache: No issues found
    ✓ Php: No issues found
    ✓ Resources: No issues found
    ? Connectivity: Found possible issues
    ✓ Mariadb: No issues found
 

 
[Wordpress]

Found recent WordPress plugin related error messages in the Apache error log.
```
[Wed Mar 27 01:18:57.698987 2024] [proxy_fcgi:error] [pid 31127:tid 
140397061838528] [client **ip_address**:63296] AH01071: Got error 'PHP message: 
PHP Deprecated: Creation of dynamic property ElementorPro\Plugin::$updater is 
deprecated in /bitnami/wordpress/wp-content/plugins/elementor-pro/plugin.php on 
line 452; PHP message: PHP Deprecated: Creation of dynamic property 
Press [Enter] to continue:
Kirki\Field\Repeater::$compiler is deprecated in 
/bitnami/wordpress/wp-content/plugins/kirki/kirki-packages/compatibility/
```
Please check the following guide to deactivate plugins:
 
https://developer.wordpress.org/cli/commands/plugin/deactivate/

[Connectivity]

Server ports 22, 80 and/or 443 are not publicly accessible. Please check the 
following guide to open server ports for remote access:

https://docs.bitnami.com/general/faq/administration/use-firewall/
 

===== End of bndiagnostic tool output =====

                                    
Press [Enter] to continue:
The diagnostic bundle was uploaded successfully to the Bitnami servers. Please copy the following code:

### bndiagnostic was not useful. Could you please tell us why?

Did not appear to reveal any error associated with issue

### Describe your issue as much as you can

Description of issues: 

The Cariera wordpress theme licence status is not persistent. After entering the licence key, and receiving confirmation of successful validation the license status reverts back to “unlicensed” after a period of 10minutes. 

Actions Undertaken Thus Far: 

I have made the wp-config file 664 permission. However in the logs there is no indication the them files are attempting to interact with this wp-config. 

I have uninstalled all other plugins and themes and re-attempted to licence but still the issue persists. I have even gone so far as to deploy a fresh Bitnami WordPress image on a new virtual machine. 

I have run wordpress logs while replicating the issue; there are several PHP depreciation warnings but the theme developer has insisted this is not related to the issue. 

The developer informed me of the php files that execute the licence key validation with their API. I was able to monitor the handshake between the site and their sever using netcat while performing the licencing activation so I am reasonably confident the issue is not related to firewalls or communication issues.

This is also what I observe at the front end because the licence key validation is initially successful and then later appears to spuriously deactivate after a period of ten minutes . I do feel like some background process is causing this to occur, although I have been able to find anything. 

After two weeks of this attempting to correct this issue, I have exhausted all avenues within the domain of my expertise and the theme developer is pushing my towards Bitnami support. 

I have attached;

- screenshots
- logs file from WP Debug
- logfiles from var/logs

Appreciate if you could point me towards a better method of identifying the cause or conflict which is causing this spurious de-activation. Thanks in advance. 

[Bitnami support ticket.pdf](https://github.com/bitnami/vms/files/14766449/Bitnami.support.ticket.pdf)
[debug.log](https://github.com/bitnami/vms/files/14766454/debug.log)
[logs.zip](https://github.com/bitnami/vms/files/14766528/logs.zip)

**Comment by doduffy88** [comment]
Hello, I have a small update after running some tests; 

I captured some communication packets between the website server and GNO API which is used by the Cairera base file to validate the licence (api.gnodesign.com).

I performed 3 different tests while capturing these packets;

Test 1 – Was to input an erroneous key number and observe communication

Test 2 – Was to input a correct key number and observe communication

Test 3 – To wait and observe communication when Cairea theme file executed its automatic licence check with the API (about 10min after the licence was activated manually in test 2)

The conclusion was that communication packets structure (number of exchanges) appeared similar in Test 1 and Test 3 , The erroneous key data transmission ends with the server sending a FIN after receiving the license key data. This could suggest that the server detected an issue with the license key and terminated the connection. Exactly the same behaviour is exhibited in Test 3 (automatic key validation).

Is there any reason the theme files could be sending the json payload without the correct key? Is it possible that permission configuration of the theme files located in onboarding could impact / prevent the necessary interactions between these files to obtain the key and apply it within the payload?

Is there any reason that when licence validation is initiated manually by user it is successful, but fails when perform automatically?

For your information the current permission of the onboarding files is as follow;

-rw-rw-rw- 1 daemon daemon 28097 Mar 27 10:05 cariera-base.php

-rw-rw-r– 1 daemon daemon 8876 Mar 26 23:10 license.php

-rw-rw-r– 1 daemon daemon 13319 Mar 26 23:10 onboarding.php

thanks again. 
[Licence_Validation_Communication between site and GNO API.txt](https://github.com/bitnami/vms/files/14781602/Licence_Validation_Communication.between.site.and.GNO.API.txt)
